cg/meta/transfer/external_data.py

```diff
@@ -5,210 +5,204 @@
-        Path(self.destination_path % cust).mkdir(exist_ok=True)
+        self._set_parameters(ticket=ticket, dry_run=dry_run)
+        LOG.debug(f"Transferring files for ticket {self.ticket} of customer {self.customer_id}")
+        log_dir: Path = self._create_log_dir()
+        self._get_destination_path().mkdir(exist_ok=True)
 
         command: str = RSYNC_CONTENTS_COMMAND.format(
-            source_path=self.get_source_path(customer=cust, ticket=ticket),
-            destination_path=self.get_destination_path(customer=cust),
+            source_path=self._get_source_path(),
+            destination_path=self._get_destination_path(),
         )
-        sbatch_parameters: Sbatch = Sbatch(
-            job_name=ticket + self.RSYNC_FILE_POSTFIX,
+        sbatch_parameters = Sbatch(
+            job_name=self.ticket + self.RSYNC_FILE_POSTFIX,
             account=self.account,
             number_tasks=1,
             memory=1,
             log_dir=str(log_dir),
             email=self.mail_user,
             hours=24,
             commands=command,
-            error=error_function,
+            error=ERROR_RSYNC_FUNCTION.format(),
         )
         self.slurm_api.set_dry_run(dry_run=dry_run)
-        sbatch_content: str = self.slurm_api.generate_sbatch_content(
-            sbatch_parameters=sbatch_parameters
-        )
-        sbatch_path: Path = Path(log_dir, ticket + self.RSYNC_FILE_POSTFIX + ".sh")
+        sbatch_content: str = self.slurm_api.generate_sbatch_content(sbatch_parameters)
+        sbatch_path = Path(log_dir, self.ticket + self.RSYNC_FILE_POSTFIX + ".sh")
         self.slurm_api.submit_sbatch(sbatch_content=sbatch_content, sbatch_path=sbatch_path)
-        LOG.info(
-            "The folder {src_path} is now being rsynced to hasta".format(
-                src_path=self.get_source_path(customer=cust, ticket=ticket)
-            )
+        LOG.info(f"The folder {self._get_source_path().as_posix()} is now being rsynced to hasta")
+
+    def _curate_sample_folder(self, sample_folder: Path) -> None:
+        """
+        Changes the name of the folder to the sample internal_id. If force is set to True,
+        replaces any previous folder.
+        Raises:
+            Exception if the sample from the folder is not present in statusdb.
+        """
+        # TODO: Implement dry-run
+        customer: Customer = self.status_db.get_customer_by_internal_id(self.customer_id)
+        customer_folder: Path = sample_folder.parent
+        sample: Sample = self.status_db.get_sample_by_customer_and_name(
+            customer_entry_id=[customer.id], sample_name=sample_folder.name
         )
+        if (sample and not customer_folder.joinpath(sample.internal_id).exists()) or (
+            sample and self.force
+        ):
+            sample_folder.rename(customer_folder.joinpath(sample.internal_id))
+        elif not sample and not self.status_db.get_sample_by_internal_id(sample_folder.name):
```

Comment:
Could we consider breaking out these if statements to make it a bit easier to read? Also, seems weird that we have both internal ids and sample names that need to be handled the same way...

Reply:
Good that I am not the only one (This is how the code was written originally, I just moved it). I will refactor this.

Reply:
fixed @islean! Could you give it a look to check if I missed any logic?

Reply:
Much cleaner! Only concern would be if the Path.rename method breaks if there is no change in naming (as would happen now if it was already named with the internal id). Other than that, much easier on the eyes!

Reply:
I guess that if it is working right now there is no issue to it, the behaviour is the same as before